cmd/kubectl-testkube/commands/common/helper.go

```diff
@@ -624,6 +625,88 @@ func LoginUser(authUri string, customConnector bool, port int) (string, string,
 	return idToken, refreshToken, nil
 }
 
+// extractAndCleanDomain extracts domain from email and removes dots and hyphens for connector ID
+func extractAndCleanDomain(email string) (string, error) {
+	if !strings.Contains(email, "@") || strings.Count(email, "@") != 1 {
+		return "", fmt.Errorf("invalid email format")
+	}
+
+	parts := strings.Split(email, "@")
+	if len(parts) != 2 || parts[1] == "" {
+		return "", fmt.Errorf("invalid email domain")
+	}
+
+	domain := parts[1]
+
+	cleanedDomain := strings.ToLower(domain)
+	cleanedDomain = strings.ReplaceAll(cleanedDomain, ".", "")
+	cleanedDomain = strings.ReplaceAll(cleanedDomain, "-", "")
+
+	return cleanedDomain, nil
+}
+
+// validateSSOConnector checks if SSO connector exists for the domain
+func validateSSOConnector(authUri, domain string) error {
+	if !strings.HasSuffix(authUri, "/") {
+		authUri += "/"
+	}
+
+	validateURL := fmt.Sprintf("%sauth/validate?connector_id=%s", authUri, url.QueryEscape(domain))
+
+	resp, err := http.Get(validateURL)
+	if err != nil {
+		return fmt.Errorf("failed to validate SSO connector: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("SSO not configured for domain %s (HTTP %d)", domain, resp.StatusCode)
+	}
+
+	return nil
+}
+
+func LoginUserSSO(apiUrl, authUrl, email string, port int) (string, string, error) {
+	ui.H1("SSO Login")
+
+	// Validate email format and extract domain
+	domain, err := extractAndCleanDomain(email)
+	if err != nil {
+		return "", "", fmt.Errorf("invalid email format: %w", err)
+	}
+
+	// Validate that SSO is configured for this domain using API URL
+	err = validateSSOConnector(apiUrl, domain)
+	if err != nil {
+		return "", "", fmt.Errorf("SSO validation failed: %w", err)
+	}
+
+	ui.Debug("Logging into cloud with SSO for domain", domain)
```

Comment:
```suggestion
	ui.Debug("Logging into cloud with SSO for domain ", domain)
```

I think otherwise it will get directly appended and end with `for domainxxxx`